subsys/bindesc/CMakeLists.txt

```diff
@@ -0,0 +1,75 @@
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library()
+
+if(CONFIG_X86)
+  zephyr_linker_sources(ROM_START SORT_KEY 0x1bindesc bindesc_no_vt.ld)
+  zephyr_library_sources(x86/bindesc_skip.S)
+else()
+  zephyr_linker_sources(ROM_START SORT_KEY 0x1bindesc bindesc.ld)
+endif()
+
+macro(gen_build_time_int_definition def_name format)
+  if(CONFIG_BINDESC_${def_name})
+    string(TIMESTAMP ${def_name} ${format})
+    # remove leading zeros so that the output will not be interpreted as octal
+    math(EXPR ${def_name} ${${def_name}})
+    zephyr_library_compile_definitions(${def_name}=${${def_name}})
+  endif()
+endmacro()
+
+macro(gen_build_time_str_definition def_name format)
+  if(CONFIG_BINDESC_${def_name})
+    string(TIMESTAMP ${def_name} ${${format}})
+    zephyr_library_compile_definitions(${def_name}="${${def_name}}")
+  endif()
+endmacro()
+
+macro(gen_str_definition def_name value)
+  if(CONFIG_BINDESC_${def_name})
+    zephyr_library_compile_definitions(${def_name}="${value}")
+  endif()
+endmacro()
+
+if(CONFIG_BINDESC_DEFINE_BUILD_TIME)
+  zephyr_library_sources(bindesc_build_time.c)
+  gen_build_time_int_definition(BUILD_TIME_YEAR "%Y")
+  gen_build_time_int_definition(BUILD_TIME_MONTH "%m")
+  gen_build_time_int_definition(BUILD_TIME_DAY "%d")
+  gen_build_time_int_definition(BUILD_TIME_HOUR "%H")
+  gen_build_time_int_definition(BUILD_TIME_MINUTE "%M")
+  gen_build_time_int_definition(BUILD_TIME_SECOND "%S")
+  gen_build_time_int_definition(BUILD_TIME_UNIX "%s")
+
+  gen_build_time_str_definition(BUILD_DATE_TIME_STRING
+      CONFIG_BINDESC_BUILD_DATE_TIME_STRING_FORMAT)
+  gen_build_time_str_definition(BUILD_DATE_STRING
+      CONFIG_BINDESC_BUILD_DATE_STRING_FORMAT)
+  gen_build_time_str_definition(BUILD_TIME_STRING
+      CONFIG_BINDESC_BUILD_TIME_STRING_FORMAT)
+
+  if(CONFIG_BINDESC_BUILD_TIME_ALWAYS_REBUILD)
+    add_custom_target(
+      bindesc_time_force_rebuild
+      COMMAND ${CMAKE_COMMAND} -U BUILD_TIME_YEAR ${CMAKE_BINARY_DIR}
+    )
```

Comment:
why are you trying to remove a variable from the CMake cache which never seems to be added to the cache ?

Can you provide some details on your intentions.

Reply:
@tejlmand The intention was to make cmake always rebuild `bindesc_build_time.c` so that the timestamp will be accurate. This is the solution I found somewhere, and I admit that now looking at it, `BUILD_TIME_YEAR` was not the variable name that was supposed to be there. I'm not sure if it got there by mistake in a refactor or that originally I intended for this cached value to be created for every time descriptor. Nevertheless - this solution is working. I changed `BUILD_TIME_YEAR` to `BUILD_TIME_DUMMY` (which is not set anywhere), and sure enough, when `CONFIG_BINDESC_BUILD_TIME_ALWAYS_REBUILD` is enabled it always rebuilds the file, and when it's disabled it does not.